cpmpy/solvers/pysat.py

```diff
@@ -229,17 +229,7 @@ def solve(self, time_limit=None, assumptions=None):
 
         # ensure all vars are known to solver
         self.solver_vars(list(self.user_vars))
-
-        # the user vars are only the Booleans (e.g. to ensure solveAll behaves consistently)
-        user_vars = set()
-        for x in self.user_vars:
-            if isinstance(x, _BoolVarImpl):
-                user_vars.add(x)
-            else:
-                # extends set with encoding variables of `x`
-                user_vars.update(self.ivarmap[x.name].vars())
-
-        self.user_vars = user_vars
+        self.user_vars = get_user_vars(self.user_vars, self.ivarmap)
```

Comment:
get_user_vars is a too generic name... I think `bool_user_vars` would be better aligned with its goal